adobecom/milo
# MWPW-138989 [Perf] Preload Milo Blocks if MEP enabled

When using mep with Target, an await happens that prevents any block-loading from happening until Target makes a decision. This PR preloads LCP blocks (any block that appears in section 1 on the base page) by importing the existing preloadBlockResources from the utils.js file. The PR also adds an additional parameter to preloadBlockResources to allow any block that appears in section 1 to be preloaded. 

Resolves: [MWPW-138989](https://jira.corp.adobe.com/browse/MWPW-138989)

Steps to QA:
For the before and after links, set a breakpoint around line 1076 of personalization.js (after preloadBlockResources is called).


![Screenshot 2024-09-30 at 2 02 41 PM](https://github.com/user-attachments/assets/442a2cdb-2530-45af-bed8-f8ec867e81c1)




Then open the Network tab, select either js or css and. For the After url, at this breakpoint, you should see the relevant marquee, aside and text files have loaded (along with the decorate.js file), and for the Before Url, the files should not be loaded at this point.

**Test URLs:**
- Before: https://main--cc--adobecom.hlx.page/drafts/mepdev/fragments/2024/q4/preloadblocks/testpage?martech=off
- After: https://main--cc--adobecom.hlx.page/drafts/mepdev/fragments/2024/q4/preloadblocks/testpage?milolibs=preloadblocks&martech=off

PSI-check: https://preloadblocks--milo--adobecom.hlx.page/?martech=off
